File: testdata/typescript-fixture/src/greeter.ts
// A tiny fixture module for typescript-language-server adapter integration tests.

export class Greeter {
  constructor(private readonly name: string) {}

  greet(): string {
    return `Hello, ${this.name}!`;
  }
}

export function makeGreeter(name: string): Greeter {
  return new Greeter(name);
}

File: testdata/swift-fixture/Sources/SwiftFixture/Greeter.swift
/// A tiny fixture module for sourcekit-lsp adapter integration tests.

public struct Greeter {
    public let name: String

    public init(name: String) {
        self.name = name
    }

    public func greet() -> String {
        return "Hello, \(name)!"
    }
}

public func makeGreeter(name: String) -> Greeter {
    return Greeter(name: name)
}
